### PR TITLE
dev-libs/boost: fixed mingw-w64 build

### PR DESCRIPTION
**Pull request by hanetzer**
Package-Manager: Portage-2.3.5, Repoman-2.3.2
Closes https://bugs.gentoo.org/show_bug.cgi?id=618890

### CODE REVIEW

File: dev-libs/boost/boost-1.63.0.ebuild [hunk 1]
```diff
@@ -47,6 +47,13 @@ PATCHES=(
 	"${FILESDIR}/${PN}-1.56.0-build-auto_index-tool.patch"
 	"${FILESDIR}/${PN}-1.63.0-fix-python.patch"
 )
+case ${CHOST} in
```
Comment (mgorny) [inline dev-libs/boost/boost-1.63.0.ebuild]: You aren't really supposed to check CHOST in global-scope. Add a conditional eapply in src_prepare instead.

Comment (ntzrmtthihu777) [inline dev-libs/boost/boost-1.63.0.ebuild]: Ah, ok, I'll fix that up once I get home from work, say 5h from now or so.

Comment (ntzrmtthihu777) [inline dev-libs/boost/boost-1.63.0.ebuild]: @mgorny this good?

**Comment by mgorny** [comment]
I'm looking at upstream trackers and AFAICS they didn't apply the patch but fixed the issue nevertheless. However, their commit messages are horrible and they can't find themselves what they actually did… Could you try 
if https://github.com/boostorg/serialization/commit/b0a794da38b52c8ae0f8155cf3bb30321cf67f00 might help?

**Comment by hanetzer** [comment]
@mgorny seems there is too large of a difference between `dev-libs/boost-1.63.0` and upstream to apply said commit as a patch (for instance, `include/boost/archive/impl/archive_serializer_map.ipp`
does not exist with any degree of being generous with path components. closest match is
`boost/archive/detail/archive_serializer_map.hpp`

Perhaps a newer release would work better.

**Comment by hanetzer** [comment]
@mgorny upon taking a new look at this package under mingw-w64 (dev-libs/boost-1.65.0) the same issue exists as before; the commit you specified is contained within the release.

**Comment by hanetzer** [comment]
Closing as dev-libs/boost-1.66.0 does compile and build fine under mingw-w64, with some
adjustments to the ebuild's windows-specific logic.

**Pull request closed without merging**